### DISCUSSION TITLE
Partial support for other vehicles

### DISCUSSION THREAD

**Discussion by matthieu-labas** [teslamate-org/teslamate#2694] [Ideas]
Hi!

I happen to love Teslamate *and* also have an electric motorbike where I can gather many of the same type of statistics (speed, power, charge stats, GPS position, etc.).

I wanted to add a kind of external service where I could upload those statistics as a second vehicle (though not a Tesla) but am not sure what would not break my current installation: direct insert in database or use an already-existing JSON-POSTed web service that would assess database constraints are not violated? (I guess Postgres already has those constraints implemented but I'm not sure).

Did anybody already tried such thing?

Thanks!
Regards,
Matthieu

PS: I'm *not* fluent in Elixir but in Java, C++, python or bash

**Comment by tobiasehlert** [comment]
Hi @matthieu-labas,

What type of electric motorbike do you have, where you can gather those type of data as well?

There are currently ways of importing data to TeslaMate from eg. [TeslaFI](https://docs.teslamate.org/docs/import/teslafi) and [tesla-apiscraper](https://docs.teslamate.org/docs/import/tesla_apiscraper), so theoretically it should work to import data the same way. You just have to check that the data is formatted the correct way and then it should kind of work.
**But** yeah.. I don't know what should happen to the imported data itself, since it won't have a "car" to get the data attached to, so you might need to tweak some more with it in the database or so.

If you are looking for a way to visualize data with the Grafana dashboards only, I think it would be easier to add a new datasource pointing to your electric motorbike database and add some new dashboards. Make a copy of a dashboard from TeslaMate, changed the datasource and rewrite the dashboard queries and you should end up with having some nice dashboards for your bike :)

Kind regards,
Tobias

**Reply by matthieu-labas** [reply]
Thanks for the insight @tobiasehlert! :)

My bike is an Energica and I use an app that gather live data from the bluetooth connection between the bike and the phone. Many information are the same as what Tesla has (speed, power, GPS, temperature, ...) and some others are not (e.g. battery voltage, balance). The is (usually) connected to the Internet and I can make the app push data to a small server, or have it do it later from the offline logs.

I was initially thinking of adding a new "car" to the database because I'm not as god-like as @adriankumpf  with Grafana ;) but having a separate database and an additional datasource makes more sense indeed (and requires no modification to Teslamate as well ;)).
I guess I should just directly open the Grafana "bike dashboards" then and not go through Teslamate? The more I think about it the more it seems Teslamate has nothing to do with it, I'll just borrow some of those awesome Grafana powers and "steal" some of those nice dashboards :)

Thanks again for the pointer! :+1: 

Kind regards,
Matthieu
